getting-started/CONCEPTS/APPLICATION.md

```diff
@@ -0,0 +1,69 @@
+    await context.sendActivity(`Starting sign in flow.`);
+    // start signin flow
+});
+```
+
+**C#**
+```cs
+// Listen for user to say '/login'.
+app.OnMessage("/login", async (ITurnContext turnContext, TurnState turnState, CancellationToken cancellationToken) =>
+{
+    await turnContext.SendActivityAsync("Starting sign in flow.", cancellationToken: cancellationToken);
+        // start signin flow
+});
+```
+> The `message` and `OnMessage` methods are referred to as activity or *route registration* method. 
+> The `turnContext` and `turnState` parameters are present in every route handler. To learn more about them see [TURNS](TURNS.md).
+
+The `Application` groups the route registration methods based on the specific feature groups: 
+
+
+| **Feature**       | **Description**                                                                         |
+|-------------------|-----------------------------------------------------------------------------------------|
+| Task Modules      | Task module related activities like `task/fetch`.        |
+| Message Extension | Message extension activities like `composeExtension/query`.      |
+| Meetings          | Meeting activites like `application/vnd.microsoft.meetingStart`. |
+| AdaptiveCards     | Adaptive card activities like `adaptiveCard/action`.             |
+| General           | Generic activites like `message`.                        |
+
+> To see all the route registration methods supported, see the migration docs ([JS](https://github.com/microsoft/teams-ai/blob/main/getting-started/MIGRATION/JS.md#activity-handler-methods)/[C#](https://github.com/microsoft/teams-ai/blob/main/getting-started/MIGRATION/DOTNET.md#activity-handler-methods)).
+
+In general, the activity handler system is all that is needed to have a functional bot or message extension. 
+
+## The AI System
+The AI System is an optional component used to plug in LLM powered experiences like user intent mapping, chaining...etc. It is configured once when orchestrating the application class. To learn more about it see [The AI System](./AI-SYSTEM.md).
+
+## The Routing Logic
+
+When an incoming activity reaches the server, the bot adapter handles the necessary authentication and creates a turn context object that encapsulates the activity details. Then the `Application`'s main method (`run()` in Javscript. `OnTurnAsync()` in C#) is called. It's logic can be broken down into these eight steps. 
+
+1. If configured in the application options, pulses of the `Typing` activity are sent to the user.
+2. If configured in the application options, the @mention is removed from the incoming message activity.
+3. The turn state is loaded using the configured turn state factory.
+4. If user authentication is configured, then attemp to sign the user in. If there user is already signed in, retrieve the access token and continue to step 5. Otherwise, start the sign in flow and end the current turn.
```

Comment:
```suggestion
4. If user authentication is configured, then attempt to sign the user in. If the user is already signed in, retrieve the access token and continue to step 5. Otherwise, start the sign in flow and end the current turn.
```